File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    
    sum = 0
    start = 0
    end = 0
    for i in range(len(sequence)):
        end = i
        sum += sequence[end]
        while k < sum:
            sum -= sequence[start]
            start += 1
        if k == sum:    
            if len(answer) == 0 or end - start < answer[1] - answer[0]:
                answer = [start, end]
        
    return answer

File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[] {-1, -1};
        
        int sum = 0, start = 0, end = 0;
        for(int i = 0; i < sequence.length; i++){
            end = i;
            sum += sequence[i];
            while(k < sum){
                sum -= sequence[start++];
            }
            if(k == sum){
                if((answer[0] == -1 && answer[1] == -1) || end - start < answer[1] - answer[0]){
                    answer[0] = start;
                    answer[1] = end;
                }
            }
        }
        
        return answer;
    }
}